parsl/dataflow/flow_control.py

```diff
@@ -141,8 +151,94 @@ def close(self):
         self._thread.join()
 
 
+class Timer(object):
+    '''This timer is a simplified version of the FlowControl timer.
+    This timer does not employ notify events.
+
+    This is based on the following logic :
+
+    >>> BEGIN (INTERVAL, THRESHOLD, callback) :
+    >>>     start = current_time()
+    >>>
+    >>>     while (current_time()-start < INTERVAL) :
+    >>>          wait()
+    >>>          break
+    >>>
+    >>>     callback()
+
+    '''
+
+    def __init__(self, callback, *args, interval=5):
+        ''' Initialize the flowcontrol object
+        We start the timer thread here
+
+        Args:
+             - dfk (DFK object) : DFK object to track parsl progress
+             - config (Dict) : Config dict structure
```

Comment:
`DFK object` -> `DataFlowKernel`
`Dict` -> `dict`

If you use the actual class names sphinx will cross-reference them (it may need some tweaking to the config, I will submit it as a PR).